### PR TITLE
refactor(nodebuilder/header): get rid of `InitStore`

### PR DESCRIPTION
**Pull request by renaynay**
Not needed anymore

### CODE REVIEW

File: nodebuilder/header/module_test.go [hunk 1]
```diff
@@ -36,8 +36,13 @@ func TestConstructModule_StoreParams(t *testing.T) {
...
-		fx.Provide(func() datastore.Batching {
-			return datastore.NewMapDatastore()
+		fx.Supply(modp2p.Bootstrappers{}),
+		fx.Provide(context.Background),
+		fx.Provide(libp2p.New),
+		fx.Provide(conngater.NewBasicConnectionGater),
+		fx.Provide(func() (datastore.Batching, datastore.Datastore) {
+			ds := datastore.NewMapDatastore()
+			return ds, ds
 		}),
```
Comment (renaynay) [inline nodebuilder/header/module_test.go:46]: FYI - not sure how this test worked without this to begin with..

**Review by distractedm1nd — APPROVED** [review]
utACK. Have you tested locally?

File: nodebuilder/header/module_test.go (hunk 1 above)
Comment (Wondertan) [reply nodebuilder/header/module_test.go:46]: Does it work without the change?

**Review by Wondertan — COMMENTED** [review]
My intuition says me like there might be something subtle we miss by doing so. I clearly remember we had to do it, but don't remember why. 

Does it works properly on non and initialised node node? Also, the test change is suspicious

**Comment by renaynay** [comment]
@Wondertan Yes works for inited and non-inited nodes.

**Review by Wondertan — APPROVED** [review]
It seems like there is no blocker anymore for this simplification.
Thanks for this @renaynay!

**Pull request merged**